library/src/main/scala/scala/scalajs/runtime/RuntimeLong.scala

```diff
@@ -2,52 +2,118 @@ package scala.scalajs.runtime
 
 import scala.annotation.tailrec
 
-/**
- * emulate a Java-Long using three integers.
- * taken from gwt LongLib:
- * com.google.gwt.lang.LongLib
- *
- * only used by runtime
- *
- * holds values l, m, h (low, middle, high)
- * s.t. (x.l + ((long) x.m << 22) + ((long) x.h << 44)) is equal to
- * the original value
- */
-final class RuntimeLong(
-  val l: Int,
-  val m: Int,
-  val h: Int
-) extends Number with Comparable[java.lang.Long] { x =>
+import scala.scalajs.js
+import js.|
+import js.JSNumberOps._
+import js.JSStringOps._
+
+/** Emulates a Long on the JavaScript platform. */
+final class RuntimeLong(val lo: Int, val hi: Int)
+    extends java.lang.Number with java.io.Serializable
+    with java.lang.Comparable[java.lang.Long] { a =>
 
   import RuntimeLong._
+  import Utils._
+
+  /** Constructs a Long from an Int. */
+  def this(value: Int) = this(value, value >> 31)
+
+  // Binary compatibility for the old (l, m, h) encoding
```

Comment:
That's if an old linker, say the sbt plugin 0.6.5, tries to link a codebase with a more recent version of `scalajs-library.jar` on the classpath.